components/speedreader/rust/lib/src/readability/src/util.rs

```diff
@@ -0,0 +1,32 @@
+use std::cmp::Ordering;
+
+fn elem_index<T>(v: &Vec<T>, order: Ordering) -> usize
+where
+    T: std::cmp::Ord,
+{
+    let mut min_index = 0;
+    for i in 1..v.len() {
+        let c = &v[i];
+        let min = &v[min_index];
+        if c.cmp(min) == order {
+            min_index = i;
+        }
+    }
+    min_index
+}
+
+#[inline]
+pub fn min_elem_index<T>(v: &Vec<T>) -> usize
+where
+    T: std::cmp::Ord,
+{
+    elem_index(v, Ordering::Less)
+}
+
+#[inline]
+pub fn max_elem_index<T>(v: &Vec<T>) -> usize
+where
+    T: std::cmp::Ord,
+{
+    elem_index(v, Ordering::Greater)
+}
```

Comment:
May be worth just using standard library iterator methods instead, e.g.

```rust
vec![/*...*/]
    .iter()
    .enumerate()
    .max_by_key(|(_, elm)| *elm)
    .map(|(index, _)| index)
```

Reply:
I don't really find this very readable, and the compiler has a hard time optimizing it so it generates more code that is orders of magnitude slower.

https://rust.godbolt.org/z/drfP1K